canonical-web-and-design/maas-ui
# Update analytics settings

## Done

- Update analytics settings.
- Disable usabilla when the option is turned off.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Either build this branch in a development MAAS, or update App.js to pass `<Footer debug={false}...`.
- Visit /MAAS/r/settings/configuration/general.
- Toggle the analytics on, you should see the "Give feedback" link in the footer and clicking it should show the Usabilla popup.
- Toggle it off and the link in the footer should be gone.

Note: I need to make a component update to to make the prop types warnings go away: https://github.com/canonical-web-and-design/react-components/issues/195

## Fixes

Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/2054.
